connectorx/src/transports/postgres_arrow.rs

```diff
@@ -76,6 +77,10 @@ macro_rules! impl_postgres_transport {
                 { Float4Array[Vec<Option<f32>>]      => Float32Array[Vec<Option<f32>>]         | conversion auto   }
                 { Float8Array[Vec<Option<f64>>]      => Float64Array[Vec<Option<f64>>]         | conversion auto   }
                 { NumericArray[Vec<Option<Decimal>>] => DecimalArray[Vec<Option<Decimal>>]     | conversion auto   }
+                { Vector[Vector]                     => Float32Array[Vec<Option<f32>>]         | conversion none   }
```

Comment:
`conversion option` allows you to only manually define the conversion for non-option type and the corresponding option type code will be automatically generated.

If you change `none` to `option` and remove the manual `TypeConversion` definition for the option types, will it work here for your case?